### PR TITLE
Add ``numeric_only`` support to ``DataFrame.mode``

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/tests/test_dataframe.py [hunk 1]
```diff
@@ -4254,6 +4254,21 @@ def test_idxmaxmin_empty_partitions():
...
+        {
+            "int": [1, 2, 3, 4, 5, 6, 7, 8],
+            "float": [1.0, 2.0, 3.0, 4.0, np.nan, 6.0, 7.0, 8.0],
+            "dt": [pd.NaT] + [datetime(2010, i, 1) for i in range(1, 8)],
+            "timedelta": pd.to_timedelta([1, 2, 3, 4, 5, 6, 7, np.nan]),
+        }
+    )
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    assert_eq(ddf.mode(numeric_only=False), df.mode(numeric_only=False))
```
Comment (jrbourbeau) [inline dask/dataframe/tests/test_dataframe.py:4268]: Might be good to test that we have the same default (I've not actually run the suggestion below -- am just assuming it passes)

```suggestion
    assert_eq(ddf.mode(), df.mode())
    assert_eq(ddf.mode(numeric_only=False), df.mode(numeric_only=False))
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @phofl -- one small suggestion

File: dask/dataframe/tests/test_dataframe.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/tests/test_dataframe.py:4268]: Passes locally

**Review by jrbourbeau — APPROVED** [review]
Thanks @phofl

**Pull request merged**